soh/soh/Enhancements/randomizer/randomizer_check_tracker.cpp

```diff
@@ -658,42 +598,124 @@ void CheckTrackerItemReceive(GetItemEntry giEntry) {
-            SetCheckCollected(RC_ZR_MAGIC_BEAN_SALESMAN);
-            return;
-        }
+        }*/
     }
-    auto checkArea = GetCheckArea();
-    if (gSaveContext.pendingSale != ITEM_NONE) {
-        pendingSaleCheck = true;
-        checkAreas.push_back(checkArea);
+}
+
+void CheckTrackerSceneFlagSet(int16_t sceneNum, int16_t flagType, int32_t flag) {
+    if (flagType != FLAG_SCENE_TREASURE && flagType != FLAG_SCENE_COLLECTIBLE) {
         return;
     }
-    if (scene == SCENE_DESERT_COLOSSUS && (gSaveContext.entranceIndex == 485 || gSaveContext.entranceIndex == 489)) {
-        checkAreas.push_back(RCAREA_SPIRIT_TEMPLE);
+    if (sceneNum == SCENE_GRAVEYARD && flag == 31 && flagType == FLAG_SCENE_COLLECTIBLE) { // Gravedigging tour special case
+        SetCheckCollected(RC_GRAVEYARD_DAMPE_GRAVEDIGGING_TOUR);
         return;
     }
-    if (GET_PLAYER(gPlayState) == nullptr) {
-        transitionCheck = true;
-        return;
+    for (auto [rc, rcObj] : RandomizerCheckObjects::GetAllRCObjects()) {
+        if (!IsVisibleInCheckTracker(rcObj)) {
+            continue;
+        }
+        SpoilerCollectionCheckType checkMatchType = flagType == FLAG_SCENE_TREASURE ? SpoilerCollectionCheckType::SPOILER_CHK_CHEST : SpoilerCollectionCheckType::SPOILER_CHK_COLLECTABLE;
+        SpoilerCollectionCheck scCheck = Location(rc)->GetCollectionCheck();
+        if (scCheck.scene == sceneNum && scCheck.flag == flag && scCheck.type == checkMatchType) {
+            SetCheckCollected(rc);
+            return;
+        }
+    }
+}
+
+void CheckTrackerFlagSet(int16_t flagType, int32_t flag) {
+    SpoilerCollectionCheckType checkMatchType = SpoilerCollectionCheckType::SPOILER_CHK_NONE;
+    switch (flagType) {
+        case FLAG_GS_TOKEN:
+            checkMatchType = SpoilerCollectionCheckType::SPOILER_CHK_GOLD_SKULLTULA;
+            break;
+        case FLAG_EVENT_CHECK_INF:
+            if ((flag == EVENTCHKINF_CARPENTERS_FREE(0) || flag == EVENTCHKINF_CARPENTERS_FREE(1) ||
+                 flag == EVENTCHKINF_CARPENTERS_FREE(2) || flag == EVENTCHKINF_CARPENTERS_FREE(3)) 
+                && GET_EVENTCHKINF_CARPENTERS_FREE_ALL()) {
+                SetCheckCollected(RC_GF_GERUDO_MEMBERSHIP_CARD);
+                return;
+            }
+            checkMatchType = SpoilerCollectionCheckType::SPOILER_CHK_EVENT_CHK_INF;
+            break;
+        case FLAG_INF_TABLE:
+            if (flag == INFTABLE_190) {
+                SetCheckCollected(RC_GF_HBA_1000_POINTS);
+            } else if (flag == INFTABLE_11E) {
+                SetCheckCollected(RC_GC_ROLLING_GORON_AS_CHILD);
+            } else if (flag == INFTABLE_GORON_CITY_DOORS_UNLOCKED) {
+                SetCheckCollected(RC_GC_ROLLING_GORON_AS_ADULT);
+            } else if (flag == INFTABLE_139) {
+                SetCheckCollected(RC_ZD_KING_ZORA_THAWED);
+            } else if (flag == INFTABLE_191) {
+                SetCheckCollected(RC_MARKET_LOST_DOG);
+            }
```

Comment:
Because I don't know how to translate the `SpoilerCheckCollectionCheck::InfTable` type into the values I actually get from the hook. IF you could enlighten me, I would be more than willing to put them in there.

Reply:
It's the reverse of the bit math you'll find in the Flags_Set functions, but since there's only 5 of these anyway, it's probably easiest to just update `SpoilerCheckCollectionCheck::InfTable` to take one argument that is the flag we use.

Reply:
Sounds like a v3 thing, though, right?

Reply:
Yeah thats fine